### PR TITLE
Async/await for util scripts

### PR DESCRIPTION
**Pull request by ChrisAntaki**
Updates some util scripts with async/await where it helps

### CODE REVIEW

File: src/utils/string.js [hunk 1]
```diff
@@ -206,7 +206,6 @@ export function hash(stringToHash) {
     return Promise.reject(message);
   }
 
-  return subtle
-    .digest('SHA-512', utf8EncodeSync(stringToHash))
-    .then((digest) => toHex(digest));
+  const digest = await subtle.digest('SHA-512', utf8EncodeSync(stringToHash));
+  return toHex(digest);
```
Comment (hartmannr76) [inline src/utils/string.js:210]: Can you not just inline it?

```js
return toHex(await subtle.digest('SHA-512', utf8EncodeSync(stringToHash)));`
```

File: src/utils/xhr.js [hunk 2]
```diff
@@ -320,21 +325,9 @@ export class FetchResponse {
...
-      this.drainText_().then(parseJson)
-    );
-  }
-
-  /**
-   * Drains the response and returns a promise that resolves with the response
-   * ArrayBuffer.
-   * @return {!Promise<!ArrayBuffer>}
-   */
-  arrayBuffer() {
```
Comment (hartmannr76) [inline src/utils/xhr.js:334]: Unused?

**Review by hartmannr76 — APPROVED** [review]

File: src/utils/xhr.js (hunk 2 above)
Comment (ChrisAntaki) [reply src/utils/xhr.js:334]: Yes... also it seems to return a Uint8Array instead of an ArrayBuffer

Comment (ChrisAntaki) [reply src/utils/xhr.js:334]: I'll rollback the changes to XHR for this PR. I'm seeing some opportunities to remove polyfills

**Pull request merged**